app/src/main/java/de/westnordost/streetcomplete/quests/seating/AddSeating.kt

```diff
@@ -0,0 +1,43 @@
+package de.westnordost.streetcomplete.quests.seating
+
+import de.westnordost.streetcomplete.R
+import de.westnordost.streetcomplete.data.osm.mapdata.Element
+import de.westnordost.streetcomplete.data.osm.mapdata.MapDataWithGeometry
+import de.westnordost.streetcomplete.data.osm.mapdata.filter
+import de.westnordost.streetcomplete.data.osm.osmquests.OsmFilterQuestType
+import de.westnordost.streetcomplete.data.osm.osmquests.Tags
+import de.westnordost.streetcomplete.data.user.achievements.QuestTypeAchievement
+import de.westnordost.streetcomplete.ktx.toYesNo
+
+class AddSeating : OsmFilterQuestType<Seating>() {
+    override val elementFilter = """
+        nodes, ways with amenity ~ restaurant|cafe|fast_food|ice_cream|food_court|pub|bar
+        and takeaway != only
+        and (!outdoor_seating or !indoor_seating)
+    """
+    override val changesetComment = "Add seating info"
+    override val defaultDisabledMessage = R.string.default_disabled_msg_summer_outdoor_seating
+    override val wikiLink = "Key:outdoor_seating"
+    override val icon = R.drawable.ic_seating_quest
+    override val isReplaceShopEnabled = true
+
+    override val questTypeAchievements = listOf(QuestTypeAchievement.CITIZEN)
+
+    override fun getTitle(tags: Map<String, String>) = R.string.quest_seating_name_title
+
+    override fun getHighlightedElements(element: Element, getMapData: () -> MapDataWithGeometry) =
+        getMapData().filter("""
```

Comment:
I think that are shoplike objects should be displayed here (see how it looks in other shop-related quests) to give a better orientation info.

Reply:
I am not sure to understand, I don't think a lot of shops in the list : https://wiki.openstreetmap.org/wiki/Key:shop have seating available. IMO, It may add too many quests

Reply:
Yeah I don't think we should ask on shop=* as they are mostly retail stores to purchase goods or services. I can't even think of a single one where we should ask.